onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -773,32 +793,61 @@ OrtStatus* PopulateTensorWithData<std::string>(OrtValue* oval, const std::string
+  return st ? st : PopulateTensorWithData<TensorElemType>(*out, tensor_data, shape.Size());
+}
+
 template <typename T>
-OrtStatus* OrtGetValueImplSeqOfPrimitives(const OrtValue* p_ml_value, int index, OrtAllocator* allocator,
-                                          OrtValue** out) {
-  using ElemType = typename T::value_type;
+OrtStatus* OrtGetValueImplSeqOfTensors(const OrtValue* p_ml_value, int index, OrtAllocator* allocator,
+                                       OrtValue** out) {
   auto& data = p_ml_value->Get<T>();
-  auto& data_elem = data.at(index);
-  std::vector<int64_t> dims = {1};
-  OrtStatus* st = OrtApis::CreateTensorAsOrtValue(allocator, dims.data(), dims.size(),
-                                                  GetONNXTensorElementDataType<ElemType>(), out);
-  return st ? st : PopulateTensorWithData<ElemType>(*out, &data_elem, 1);
+  auto& one_tensor = data.tensors.at(index);
+
+  auto tensor_elem_type = one_tensor.DataType();
+  OrtStatus* st{};
+  if (tensor_elem_type == DataTypeImpl::GetType<bool>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<bool>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<float>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<float>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<MLFloat16>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<MLFloat16>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<double>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<double>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<int8_t>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<int8_t>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<uint8_t>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<uint8_t>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<int16_t>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<int16_t>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<uint16_t>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<uint16_t>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<int32_t>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<int32_t>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<uint32_t>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<uint32_t>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<int64_t>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<int64_t>(allocator, one_tensor, out);
+  } else if (tensor_elem_type == DataTypeImpl::GetType<std::string>()) {
+    st = OrtGetValueImplSeqOfTensorsHelper<std::string>(allocator, one_tensor, out);
+  } else {
+    st = OrtApis::CreateStatus(ORT_FAIL, "Only sequences that contain float tensors are supported.");
+  }
```

Comment:
Is this error message correct?

Reply:
Ah, crap! Thanks for identifying. I'm writing the code for a few other Sequence ops and will fix it as part of the follow up PR.